src/components/figures/chart/chartJs/chartjs_waterfall_plugin.ts

```diff
@@ -0,0 +1,70 @@
+import { ChartType, Plugin } from "chart.js";
+
+interface WaterfallPluginOptions {
+  showConnectorLines: boolean;
+}
+
+declare module "chart.js" {
+  interface PluginOptionsByType<TType extends ChartType> {
+    waterfallLinesPlugin?: WaterfallPluginOptions;
+  }
+}
+
+export const waterfallLinesPlugin: Plugin = {
+  id: "waterfallLinesPlugin",
+  beforeDraw(chart, args, options: WaterfallPluginOptions) {
+    if (!options.showConnectorLines) {
+      return;
+    }
+    // Note: private properties are not in the typing of chartJS (and some of the existing types are missing properties)
+    // so we don't type anything in this file
+    const drawData = (chart as any)._metasets?.[0]?.data;
+    if (!drawData) {
+      return;
+    }
+    const ctx = chart.ctx;
+    ctx.save();
+    const startIndex = drawData.findIndex((bar: any) => bar.height !== 0);
+    if (startIndex === -1) {
+      return;
+    }
+    let bar = drawData[startIndex];
+    let nextBar = getNextNonEmptyBar(drawData, startIndex);
+
+    while (nextBar) {
```

Comment:
I'm not sure it would be a better approach, but can't we do this with something like : 
```js
for (const [i, nextBar] of drawData.entries()) {
  if (i === 0 || nextBar.height === 0) {
    continue;
  }
  ...
}
```
 ?
It will avoid using a while loop with `drawData.find(...)` each time